### PR TITLE
fix: url parsing for non git urls: oci://, no protocol  (#11816)

### LINKED ISSUE #11816: 2.6 RC1 - 'URL parsing failed' error when opening application sourced from OCI registry (ECR)
**Describe the bug**

I am using the latest nightly build (quay.io/argoproj/argocd@sha256:3e3890cd3614fff9d86db4facd99349cad0241fee7f974fefdd97fdca8c13ef2). Same happens with [v2.6.0-rc1](https://github.com/argoproj/argo-cd/releases/tag/v2.6.0-rc1).

All my applications from the ECR OCI registry fail to open with an error "**URL parsing failed**".
It seems to me that it expects to have a protocol in the repoURL, because if I use - "**repoURL: 1234567890.dkr.ecr.us-west-1.amazonaws.com**" - the UI crashes. When I add "**oci://**" or "**https://**" - the UI works but, of course, the application is in an unknown state due to the "**unsupported protocol scheme "oci"**".

**To Reproduce**

Use OCI registry without protocol in the URI

**Expected behavior**

Either allow adding "oci://" prefix to the repoURL or support URLs without a protocol

**Screenshots**

image

**Version**

```json
{
    "Version": "v2.6.0+43c928e",
    "BuildDate": "2022-12-21T19:47:17Z",
    "GitCommit": "43c928e2e5f474f8e78f66f73cd0ee2bf91a0cd1",
    "GitTreeState": "clean",
    "GoVersion": "go1.18.9",
    "Compiler": "gc",
    "Platform": "linux/arm64",
    "KustomizeVersion": "v4.5.7 2022-08-02T16:35:54Z",
    "HelmVersion": "v3.10.3+g835b733",
    "KubectlVersion": "v0.24.2",
    "JsonnetVersion": "v0.19.1"
}
```

### PR DESCRIPTION
**Pull request by alexef**
fixes: #11816 
Screenshot 2022-12-23 at 10 03 10
Screenshot 2022-12-23 at 10 03 39

Signed-off-by: Alex Eftimie <alex.eftimie@getyourguide.com>

Note on DCO:

If the DCO action in the integration test fails, one or more of your commits are not signed off. Please click on the *Details* link next to the DCO action for instructions on how to resolve this.

Checklist:

* [ ] Either (a) I've created an [enhancement proposal](https://github.com/argoproj/argo-cd/issues/new/choose) and discussed it with the community, (b) this is a bug fix, or (c) this does not need to be in the release notes.
* [ ] The title of the PR states what changed and the related issues number (used for the release note).
* [ ] I've included "Closes [ISSUE #]" or "Fixes [ISSUE #]" in the description to automatically close the associated issue.
* [ ] I've updated both the CLI and UI to expose my feature, or I plan to submit a second PR with them.
* [ ] Does this PR require documentation updates?
* [ ] I've updated documentation as required by this PR.
* [ ] Optional. My organization is added to USERS.md.
* [ ] I have signed off all my commits as required by [DCO](https://github.com/argoproj/argoproj/tree/master/community#contributing-to-argo)
* [ ] I have written unit and/or e2e tests for my change. PRs without these are unlikely to be merged.
* [ ] My build is green ([troubleshooting builds](https://argo-cd.readthedocs.io/en/latest/developer-guide/ci/)).

### CODE REVIEW

**Comment by alexef** [comment]
@crenshaw-dev / @rbreeze please have a look

**Comment by RomanBats** [comment]
Looks good to me, can we merge it? 😅

File: ui/src/app/shared/components/urls.ts [hunk 1]
```diff
@@ -26,7 +26,12 @@ export function repoUrl(url: string): string {
 }
 
 export function revisionUrl(url: string, revision: string, forPath: boolean): string {
-    const parsed = GitUrlParse(url);
+    try {
+        const parsed = GitUrlParse(url);
+    } catch {
+        // URL Parsing failed
```
Comment (ashutosh16) [inline ui/src/app/shared/components/urls.ts:32]: Would you like to print error on the console log?

File: ui/src/app/shared/components/urls.ts [hunk 2]
```diff
@@ -26,7 +26,12 @@ export function repoUrl(url: string): string {
...
+        return null;
```
Comment (ashutosh16) [inline ui/src/app/shared/components/urls.ts:33]: instead of `null` should an empty string  be returned?

**Review by rbreeze — APPROVED** [review]
LGTM!

**Comment by RomanBats** [comment]
So it looks like the "parsed" constant is not visible outside the scope of try and it gets removed:
image
image

**Comment by alexef** [comment]
@RomanBats so you get this stack trace after the fix was merged? it was working fine on my machine, not sure what is going on.

**Comment by RomanBats** [comment]
@alexef, yes, I used this quay.io/argoproj/argocd@sha256:a315769f630c730485c974ff31007fa9aedf9294faf9404d858315d06afd8a18 nightly build. 
I am not very good with JS, but maybe my version is optimized and minified.

**Comment by woehrl01** [comment]
@alexef this definetly is an issue as the const value is block scoped to the try catch.

It should only fail for non oci repos now.

**Pull request merged**